app/controllers/HasPAYEController.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.FrontendAppConfig
+import identifiers.TypedIdentifier
+import models.Mode
+import models.requests.DataRequest
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{AnyContent, Result}
+import services.UserAnswersService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import utils.{Navigator, UserAnswers}
+import viewmodels.CommonFormWithHintViewModel
+import views.html.hasPAYE
+
+import scala.concurrent.Future
+
+trait HasPAYEController extends FrontendController with Retrievals with I18nSupport {
+
+  protected implicit val ec = play.api.libs.concurrent.Execution.defaultContext
```

Comment:
Don't use the global execution context as this will likely be resolved to the wrong one. Leave it abstract in this class and implement it with DI in extending classes.

Reply:
Got rid of this controller so the comment no longer applies I guess

Reply:
Though actually it does apply on the generic controller we created so I'll do it on there...

Reply:
Fixed - look on new generic controller HasReferenceNumberController